packages/fetchai/skills/tac_control_contract/helpers.py

```diff
@@ -42,15 +42,13 @@ def generate_good_id_to_name(nb_goods: int, contract: Contract) -> Dict[str, str
     Generate ids for things.
 
     :param nb_goods: the number of things.
+    :param contract: the instance of the contract
     :return: a dictionary mapping goods' ids to names.
     """
-
-    item_ids = contract.create_item_ids_based_on_nb_goods(
-        token_type=NFT, nb_goods=nb_goods
-    )
     max_number_of_digits = math.ceil(math.log10(nb_goods))
+    token_ids = contract.create_item_ids_based_on_nb_goods(NFT, nb_goods)
     string_format = "tac_good_{:0" + str(max_number_of_digits) + "}"
-    return {string_format.format(i) + "_id": str(item_ids[i]) for i in range(nb_goods)}
+    return {string_format.format(i) + "_id": token_ids[i] for i in range(nb_goods)}
```

Comment:
more pythonic
```suggestion
    return {string_format.format(token_id) + "_id": token_id for token_id in token_ids}
```

Reply:
You want the token_id as a name ? 
```
    def generate_id(self, token_id, item_id):
        token_id = token_id
        index = item_id
        final_id_int = (token_id << 128) + index
        return final_id_int
```

Reply:
my suggestion is just about accessing a list correctly in a for loop @Totoual